src/v/iceberg/BUILD

```diff
@@ -55,6 +55,7 @@ redpanda_cc_library(
         "datatypes.h",
     ],
     include_prefix = "iceberg",
+    visibility = ["//visibility:public"],
```

Comment:
Probably could lock these down to just `//src/v/datalake:__subpackages__`, but we're not very strict about this today, so it's fine for now 👍

Reply:
yeah, i was thinking about this, we will have to go over the source and fine tune the visibility policies